sycl/source/handler.cpp

```diff
@@ -468,6 +468,13 @@ void handler::processArg(void *Ptr, const detail::kernel_param_kind_t &Kind,
 void handler::processArg(void *Ptr, const detail::kernel_param_kind_t &Kind,
                          const int Size, const size_t Index, size_t &IndexShift,
                          bool IsKernelCreatedFromSource, bool IsESIMD) {
+
+  if (!IsESIMD &&
```

Comment:
It would seem more logical to me to move this check and diagnostics up the call chain, e.g. to `handler::extractArgsAndReqsFromLambda` or `handler.hpp/StoreLambda` to detect the problem earlier and more reliably (in case `processArgs` is not called for some cases of ESIMD kernel).

Reply:
Yep